src/FoldersView/FoldersListView.vala

```diff
@@ -34,16 +34,9 @@ public class Mail.FoldersListView : Gtk.ScrolledWindow {
         source_list = new Granite.Widgets.SourceList ();
         add (source_list);
         var session = Mail.Backend.Session.get_default ();
-        foreach (var account in session.get_accounts ()) {
-            var account_item = new Mail.AccountSourceItem (account);
-            source_list.root.add (account_item);
-        }
-
-        session.account_added.connect ((account) => {
-            var account_item = new Mail.AccountSourceItem (account);
-            source_list.root.add (account_item);
-        });
+        session.get_accounts ().foreach ((account) => {add_account (account); return true;});
```

Comment:
I think I'd like to see this broken out into a separate `foreach` statement again, or at least onto more than one line. Just feels like there's a lot squeezed onto one line.